proptest/src/test_runner/runner.rs

```diff
@@ -210,6 +212,96 @@ where
     })
 }
 
+#[cfg(feature = "handle-panics")]
+mod panicky {
```

Comment:
given the unsafe usage, this section could use a bit more documentation i think. possibly specifying what the responsibility of each static variable and each function as well as clarifying the order of operations. i don't think it needs to be long but just enough to provide more clarity